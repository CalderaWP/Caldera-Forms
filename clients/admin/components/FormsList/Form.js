import React from 'react';
import {PureComponent} from 'react';
import propTypes from 'prop-types';
import {ShortcodeViewer} from "./ShortcodeViewer";
import classNames from 'classnames';
//import {Button} from '@wordpress/components'

/**
 * Show one form in the FormList
 */
export class Form extends PureComponent {

	/**
	 * Create Form component
	 *
	 * @param props
	 */
	constructor(props) {
		super(props);
		this.state = {
			showShortcode: false,
		};

		this.toggleShortcodeView = this.toggleShortcodeView.bind(this);
		this.getEntriesCount = this.getEntriesCount.bind(this);
		this.openEntryViewerForForm = this.openEntryViewerForForm.bind(this);
	}

	/**
	 * Open or close the shortcode view
	 */
	toggleShortcodeView() {
		this.setState({showShortcode: !this.state.showShortcode});
	}

	/**
	 * Get the entries count
	 *
	 * Returns false if disabled
	 * @return {Number|Boolean}
	 */
	getEntriesCount() {
		if (!this.props.form.hasOwnProperty('entries')|| !this.props.form.entries.hasOwnProperty('count')) {
			return false;
		}
		return parseInt(this.props.form.entries.count,10);
	}

	/**
	 * Dispatch action to open entry viewer for one form
	 */
	openEntryViewerForForm() {
		this.props.openEntryViewerForForm(this.props.form.ID);
	}

	/**
	 * Render Form list item
	 * @return {*}
	 */
	render() {

		const activeForm = this.props.form.hasOwnProperty('form_draft')
			? this.props.form.form_draft
			: false;
		return (

			<tr
				id={`form_row_${this.props.form.ID}`}
				className={classNames({
					alternate: this.props.isAlternate,
					form_entry_row: true,
				})}
			>
				<td
					className={
						classNames(
							{'active-form': activeForm}
						)
					}
				>
					{!this.state.showShortcode &&
					<span className="cf-form-name-preview">{this.props.form.name}</span>
					}
					<span className="cf-form-view-shorcode">
						<ShortcodeViewer
							formId={this.props.form.ID}
							onButtonClick={this.toggleShortcodeView}
							show={this.state.showShortcode}
						/>
					</span>

					<span className="row-actions">
						<span className="edit">
							<a
								href={`${this.props.form.editLink}`}
							>
								Edit
							</a>
							{false !== this.getEntriesCount() &&
								<button
									className={ Form.classNames.entryButton }
									onClick={this.openEntryViewerForForm}
								>
									View Entries
								</button>
							}

						</span>
					</span>
				</td>
				<td
					style={
						{
							width: '4em',
							textAign: 'center'
						}
					}
					className={`entry_count_${this.props.form.ID}`}
				>
					{false === this.getEntriesCount() &&
					<React.Fragment>
						Disabled
					</React.Fragment>
					}
					{false !== this.getEntriesCount() &&
					<React.Fragment>
						{this.getEntriesCount()}
					</React.Fragment>
					}
				</td>
			</tr>

		);
	}

}

/**
 * Prop definitions for form component
 *
 * @type {{form: *, onFormUpdate: *, openEntryViewerForForm: shim}}
 */
Form.propTypes = {
	isAlternate: propTypes.bool,
	form: propTypes.object.isRequired,
	onFormUpdate: propTypes.func.isRequired,
	openEntryViewerForForm: propTypes.func.isRequired
};


Form.classNames = {
	isAlternate: false,
	entryButton: 'view-entry-button'
};


