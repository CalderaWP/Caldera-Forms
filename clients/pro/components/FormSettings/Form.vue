<template>
	<div>
		<div class="caldera-config-group">
			<label v-if="form.send_local != true" v-bind:for="sendLocalIdAttr">
					Disable enhanced delivery for this form
			</label>
			<label v-else v-bind:for="sendLocalIdAttr">
					Enhanced delivery is disabled for this form
			</label>
			<div class="caldera-config-field">
				<input
						type="checkbox"
						v-model="form.send_local"
						v-bind:id="sendLocalIdAttr"
						@change="changeSendLocal"
				/>
			</div>
		</div>
		<div v-if="form.send_local != true">
			<div class="caldera-config-group">
				<label v-bind:for="layoutIdAttr">
					Email Layout
				</label>
				<div class="caldera-config-field">
					<select
						v-bind:id="layoutIdAttr"
						v-model="form.layout"
						@change="changeLayout"
					>
						<option></option>
						<option v-for="option in layouts" v-bind:value="option.id">
							{{ option.name }}
						</option>
					</select>
				</div>
			</div>
			<div class="caldera-config-group">
				<label v-bind:for="pdfLayoutIdAttr">
					PDF Layout
				</label>
					<div class="caldera-config-field">
						<select
								v-bind:id="pdfLayoutIdAttr"
								v-model="form.pdf_layout"
								@change="changePDFLayout"
						>
							<option></option>
							<option v-for="option in layouts" v-bind:value="option.id">
								{{ option.name }}
							</option>
						</select>
					</div>
			</div>
			<div class="caldera-config-group">
				<label v-bind:for="attachPDFIdAttr">
					Attach PDF To Main Mailer
				</label>
				<div class="caldera-config-field">
					<input
						type="checkbox"
						v-model="form.attach_pdf"
						v-bind:id="attachPDFIdAttr"
						@change="changeAttachPDF"
					/>
				</div>
			</div>
			<div class="caldera-config-group">
				<label v-bind:for="attachPDFIdAttr">
					Add PDF Link
				</label>
				<div class="caldera-config-field">
					<input
						type="checkbox"
						v-model="form.pdf_link"
						v-bind:id="attachPDFIdAttr"
						@change="changeLinkPDF"
					/>
				</div>
			</div>
		</div>
	</div>
</template>
<script>
	import { mapState } from 'vuex'
	import Checkbox from '../Elements/Field/Checkbox';
	import { findForm } from '../../store/util/utils';

	export default{
		components :{
			checkbox: Checkbox
		},
		props : [ 'form', 'layouts' ],
		computed :{
			layoutIdAttr(){
				return 'cf-pro-layout-' + this.form.form_id;
			},
			pdfLayoutIdAttr(){
				return 'cf-pro-layout-pdf-' + this.form.form_id;
			},
			attachPDFIdAttr(){
				return 'cf-pro-layout-' + this.form.form_id;
			},
			linkPDFIdAttr(){
				return 'cf-pro-layout-' + this.form.form_id;
			},
            sendLocalIdAttr(){
                return 'cf-pro-send-local-' + this.form.form_id;
            },
		},
		methods:{
			commitChange(what,value){
				this.form[what] = value;
				this.$store.commit( 'form', this.form );
			},
			changeLayout(ev){
				this.commitChange(ev.target.value,'layout');
			},
			changePDFLayout(ev){
				this.commitChange(ev.target.value,'pdf_layout');
			},
			changeAttachPDF(ev){
				this.commitChange(ev.target.value,'attach_pdf');
			},
			changeLinkPDF(ev){
				this.commitChange(ev.target.value,'pdf_link');
			},
            changeSendLocal(ev){
                this.commitChange(ev.target.value,'send_local');
            }
		}


	}
</script>