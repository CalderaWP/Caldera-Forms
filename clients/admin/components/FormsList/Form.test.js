import renderer from "react-test-renderer";
import {Form} from "./Form";
import React from 'react';
import {shallow, mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});
const formOne = {
	ID: 'cf1',
	name: 'Form One',
	editLink: '/forms/edit/1'
};

describe('Forms component', () => {

	it('Shows form', () => {
		const formList = renderer.create(
			<Form
				form={formOne}
				onFormUpdate={() => {
				}}
				openEntryViewerForForm={() => {
				}}
			/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});


	it('Passes form ID to openEntryViewerForForm ', () => {
		let id = '';
		const component = mount(
			<Form
				form={{
					...formOne,
					entries: {
						count: 4
					}
				}}
				onFormUpdate={() => {
				}}
				openEntryViewerForForm={(formID) => {
					id = formID;
				}}
			/>
		);

		component.find('.view-entry-button').simulate('click');
		expect(id).toEqual(formOne.ID);
	});

	it('Uses alternate class when prop is passed', () => {
		//alternate

		const component = mount(
			<Form
				form={formOne}
				onFormUpdate={() => {
				}}
				openEntryViewerForForm={() => {
				}}
				isAlternate={true}
			/>
		);
		expect(component.find('.alternate').length).toBe(1);
	});

	it('Does not use alternate class when prop is not passed', () => {
		//alternate

		const component = mount(
			<Form
				form={formOne}
				onFormUpdate={() => {
				}}
				openEntryViewerForForm={() => {
				}}
			/>
		);
		expect(component.find('.alternate').length).toBe(0);
	});

	it('Does not use alternate class when prop is false', () => {
		//alternate

		const component = mount(
			<Form
				form={formOne}
				onFormUpdate={() => {
				}}
				openEntryViewerForForm={() => {
				}}
				isAlternate={false}
			/>
		);
		expect(component.find('.alternate').length).toBe(0);
	});
});