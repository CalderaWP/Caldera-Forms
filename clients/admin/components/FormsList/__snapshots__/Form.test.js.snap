// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms component Shows form 1`] = `
<tr
  className="form_entry_row"
  id="form_row_cf1"
>
  <td
    className=""
  >
    <span
      className="cf-form-name-preview"
    >
      Form One
    </span>
    <span
      className="cf-form-view-shorcode"
    >
      <button
        className="button cf-form-shortcode-preview-button"
        onClick={[Function]}
      >
        Get Shortcode
      </button>
    </span>
    <span
      className="row-actions"
    >
      <span
        className="edit"
      >
        <a
          href="/forms/edit/1"
        >
          Edit
        </a>
      </span>
    </span>
  </td>
  <td
    className="entry_count_cf1"
    style={
      Object {
        "textAign": "center",
        "width": "4em",
      }
    }
  >
    Disabled
  </td>
</tr>
`;
